js-exercises/deep-equal/deepEqual.js

```diff
@@ -0,0 +1,77 @@
+function deepEqual(firstObj, secondObj, matchDescriptors = false) {
+  if (typeof matchDescriptors !== 'boolean') {
+    throw new TypeError(`Expected third argument as a boolean, got ${typeof matchDescriptors}`);
+  }
+  if (arguments.length <= 1) {
+    throw new TypeError(`Expected at least two arguments for compare, got ${arguments.length}`);
+  }
+  return checkObjectEquality(firstObj, secondObj, matchDescriptors);
+}
+
+function hasSameType(firstObj, secondObj) {
+  return typeof firstObj === typeof secondObj;
+}
+
+function checkObjectEquality(firstObj, secondObj, matchDescriptors) {
+  if (!hasSameType(firstObj, secondObj)) return false;
+  if (typeof firstObj !== 'object') {
+    if (typeof firstObj === 'function') return firstObj.toString() === secondObj.toString();
```

Comment:
`(typeof firstObj === 'function') return firstObj.toString() === secondObj.toString();`

This would be true for most functions. You need to check them like objects.